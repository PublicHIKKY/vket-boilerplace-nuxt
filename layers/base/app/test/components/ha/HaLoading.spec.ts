import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'
import HaLoading from '#base/app/components/ha/HaLoading.vue'

test('ref component', () => {
  expect(HaLoading).toBeTruthy()
})

test('mount component', () => {
  const wrapper = mount(HaLoading)
  expect(wrapper.getCurrentComponent()).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()
})

describe(':manual', () => {
  it('no :manual make no component on mount', () => {
    const wrapper = mount(HaLoading)
    expect(wrapper.isVisible()).toBeFalsy()
  })
  it(':manual="true" make spinner', () => {
    const wrapper = mount(HaLoading, {
      props: {
        manual: true,
      },
    })
    expect(wrapper.isVisible()).toBeTruthy()
    expect(wrapper.find('.spinner').exists()).toBeTruthy()
  })
})

describe(':cover', () => {
  it(':cover make <div class="cover">', () => {
    const wrapper = mount(HaLoading, {
      props: {
        manual: true,
        cover: true,
      },
    })
    expect(wrapper.find('.cover').exists()).toBeTruthy()
  })
})
