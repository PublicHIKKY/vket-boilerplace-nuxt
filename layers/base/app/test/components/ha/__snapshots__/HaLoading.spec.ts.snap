// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `"<!--v-if-->"`;
